lib/puppet/parser/functions/enclose_ipv6.rb

```diff
@@ -28,16 +28,18 @@ module Puppet::Parser::Functions
     result = []
 
     input.each do |val|
-      begin
-        ip = IPAddr.new(val)
-      rescue *rescuable_exceptions
-        raise(Puppet::ParseError, "enclose_ipv6(): Wrong argument "+
-          "given #{val} is not an ip address.")
+      unless val == '*'
+        begin
+          ip = IPAddr.new(val)
+        rescue *rescuable_exceptions
+          raise(Puppet::ParseError, "enclose_ipv6(): Wrong argument "+
+                "given #{val} is not an ip address.")
+        end
+        val = "[#{ip.to_s}]" if ip.ipv6?
       end
-      val = "[#{ip.to_s}]" if ip.ipv6?
-      result = [result,val]
+      result << val
     end
 
-    return result.flatten.compact
```

Comment:
It would be appreciated if you could explain why it is ok that this can be removed, and how it related to wildcard vhosts, thanks!